scene/2d/area_2d.cpp

```diff
@@ -166,6 +166,17 @@ void Area2D::_body_inout(int p_status, const RID &p_body, ObjectID p_instance, i
 	bool body_in = p_status == PhysicsServer2D::AREA_BODY_ADDED;
 	ObjectID objid = p_instance;
 
+	// Exit early if instance is invalid.
+	if (objid.is_null()) {
+		// Emit the appropriate signals.
+		if (body_in) {
+			emit_signal(SceneStringNames::get_singleton()->body_shape_entered, p_body, (Node *)nullptr, p_body_shape, p_area_shape);
+		} else {
+			emit_signal(SceneStringNames::get_singleton()->body_shape_exited, p_body, (Node *)nullptr, p_body_shape, p_area_shape);
+		}
```

Comment:
Seems to be we should do the lock-unlock thing here as well.

Reply:
Will take a look, good idea